lib/game.rb

```diff
@@ -0,0 +1,73 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :cells
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2], # top row
+    [3, 4, 5], # middle row
+    [6, 7, 8], # bottom row
+    [0, 3, 6], # left column
+    [1, 4, 7], # middle column
+    [2, 5, 8], # right column
+    [0, 4, 8], # left diag
+    [2, 4, 6] # right diag
+  ]
+
+  def initialize(player_1 = Human.new('X'), player_2 = Human.new('O'), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    @board.turn_count.even? ? @player_1 : @player_2
+    end
+
+  def over?
+      (@board.full? || won? || draw?) ? true : false
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |combination|
+      next unless @board.get_cells(combination).uniq.length == 1
+      token = @board.cells[combination[0]]
+      if (token == 'X') || (token == 'O')
+        @winner = token
+        return true
+      end
+    end
+    false
```

Comment:
You should never `return` from within a loop. If you need a return value from within a loop you need to use a higher level iterator or some sort of local variable. 

Consider this solution to a simpler problem of finding the even number in an array.

```ruby
the_even_number = nil
[1,2,3].each do |i|
  if i.even?
    the_even_number = i
  end
end
the_even_number
```

Instead of returning from within that loop, we flip the switch local variable `the_even_number` to the element in the array that made the if statement true. Then we leave `the_even_number` as the last line to ensure the return value. If no even numbers are present, `the_even_number` will remain nil.

But better, let's just use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect)

```ruby
the_even_number = [1,2,3].detect do |i|
  i.even?
end
```